Components: Move product icons from My Jetpack into components package

oskosk opened the pull request:
Moves product icons into JS components

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Moves product icons into components making them available as imports from `@automattic/jetpack-components`.
* Adds README.md to the icon directory

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Checkout this branch, on a connected site
* Visit My Jetpack
* Confirm the icons are render on the product cards.

oskosk committed:
[not verified] Move product icons from My Jetpack into components package

oskosk committed:
[not verified] changelog

oskosk committed:
[not verified] Fix indentation on readme

oskosk commented on projects/js-packages/components/components/product-icons/README.md:
```diff
@@ -0,0 +1,72 @@
+	    <StarIcon />
+        <VideoPressIcon />
+    </div>
+)
```
Addressd in 80a7205fe0e6930fb93d1988fce5aaa72421d0ca

oskosk committed:
Update package versions

retrofox reviewed (commented):
Moving the icons without any significant changes could make the API ambiguos.
Maybe we should wither rename `getIconBySlug()` by `getProductIconBySlug()`, or rename the icon slug for each icon? For Instead, rename `backup` to `product-backup`.

renatoagds commented:
I thought to move this and create a unique `Icon` component at `jetpack-components`, which could make us use it in all places:

```
import { Icon } from '@automattic/jetpack-components'
<Icon icon="boost" />
```

I think we can move the current way, and improve next.

retrofox commented:
> I thought to move this and create a unique `Icon` component at `jetpack-components`, which could make us use it in all places:
> 
> ```
> import { Icon } from '@automattic/jetpack-components'
> <Icon icon="boost" />
> ```
> 
> I think we can move the current way, and improve next.

I did an approach to follow a similar core pattern, but it lead me to think that we may should create a new pkg:

```es6
import { Icon, boost } from '@automattic/jetpack-icons'

function MyCmp() {
  return(
    <Icon icon={ boost } />
  );
}
```

oskosk commented:
@retrofox how can I reproduce that ambiguity?

renatoagds commented:
> I did an approach to follow a similar core pattern, but it lead me to think that we may should create a new pkg.

Yep, that's a way to follow. But core already has an `Icon` component that accepts a string: https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/icon#with-a-dashicon

I vote to follow with everything at `jetpack-components`, for me is really confusing to have two packages to go through, every time I got lost 😂

oskosk committed:
Merge branch 'master' into update/move-my-jetpack-product-icons-into-components

oskosk committed:
Fix test

oskosk committed:
Fix duplicate import

oskosk commented:
I've addressed some issues. @renatoagds @retrofox could you approve or reject this one? I'd like to be able to use this icons in #23586 without resorting to pure webpack path resolution

oskosk commented:
Thanks!

oskosk commented:
r242549-wpcom

oskosk commented:
The diff for this PR was problematic. As the PR moved files from My Jetpack into components there was no trace in this diff about the moved files. my-jetpack files are ignored by fusion so the diff didn't care, but it did include the README that was being introduced only in the components version of the dir. cc @Automattic/ground-control

anomiex commented:
Moved files are one of the annoying edge cases in Fusion. Sounds like you worked it out correctly.

We're working on getting rid of Fusion entirely by committing the built Jetpack extension as a whole to wpcom weekly. See p9dueE-4mF-p2.
